hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/SqlPlanImpl.java

```diff
@@ -1044,7 +1042,13 @@ public boolean producesRows() {
         }
 
         @Override
-        public SqlResult execute(QueryId queryId, List<Object> arguments, long timeout, SqlSecurityContext ssc) {
+        public void checkPermissions(SqlSecurityContext context) {
+            // Note: it must not expose the presence of the mapping.
+            extractPermissions(rel).forEach(context::checkPermission);
```

Comment:
I do not understand the comment. This leaks information about existence of the mapping (you get information about missing permission when it exists) but we accept this at present.

Reply:
Old comment